reactioncommerce/reaction
# [WIP] (refactor): 3662 customer product grid publishing

Resolves #3662  
Impact: **breaking**  
Type: **performance|refactor**

## Issue
The customer view of the product grid was taking an incredibly long time to load or in many cases just crashing the browser tab. 

Several issues with the productGridItem component were causing slow down:
* grid items had deep and confusing nested props.
* grid items were using inherited prop callback to get values already in the component's props.
* grid items had a lot of conditional logic only needed for admin views.
* the product grid was destructuring all of its props onto each grid item, this was causing each grid item to have not only it's own product data but all the entire array of products.

Issues with the "Products" publication
* 

## Solution
@mikemurray and I have been approaching this issue by separating the customer view from the admin view. 

* Created a new customer's "Products" publication that doesn't worry about any admin needed data, attaches product variants to the top-level product and fetches the appropriate product media.
* Created a few React components that only deal with customer interaction.

## Breaking changes
The old `imports/plugins/included/product-variants/containers/productsContainer.js` has been renamed to `productsContainerAdmin.js` and a new component names `productsContainer.js` now handles which products container to load based on the user's permissions.

*You'll need the [reaction-devtool](https://github.com/reactioncommerce/reaction-devtools) installed to test this PR.*
## Testing
1. Start up a fresh RC shop and log in as an Admin.
2. Open devtools panel and load the medium data set. Once that finishes loading, load the images as well.
3. In another browser view the RC shop as a customer to see product grid loading the large data set.
4. Follow [acceptance tests](https://docs.reactioncommerce.com/reaction-docs/master/acceptance-testing) 1A + 1B to verify customer views still working.
